frontend/src/features/comparisons/Comparison.tsx

```diff
@@ -61,8 +62,50 @@ const Comparison = () => {
     useState<ComparisonRequest | null>(null);
 
   const searchParams = new URLSearchParams(location.search);
-  const videoA: string = searchParams.get('videoA') || '';
-  const videoB: string = searchParams.get('videoB') || '';
+  const uidParams: { vidA: string; vidB: string } = useMemo(() => {
+    return {
+      vidA: 'uidA',
+      vidB: 'uidB',
+    };
+  }, []);
+  const legacyParams: { vidA: string; vidB: string } = useMemo(() => {
+    return {
+      vidA: 'videoA',
+      vidB: 'videoB',
+    };
+  }, []);
+
+  // step1: try to read UIDs from the URL...
+  const uidA: string = searchParams.get(uidParams.vidA) || '';
+  const uidB: string = searchParams.get(uidParams.vidB) || '';
+
+  // step2: if they are empty, try the legacy videoA/videoB parameters
+  const videoA: string = uidA
+    ? idFromUid(uidA)
+    : searchParams.get(legacyParams.vidA) || '';
+  const videoB: string = uidB
+    ? idFromUid(uidB)
+    : searchParams.get(legacyParams.vidB) || '';
+
+  // step 3: Clean the URL by replacing legacy parameters by UIDs.
+  useEffect(() => {
+    const legacyA = searchParams.get(legacyParams.vidA);
+    const legacyB = searchParams.get(legacyParams.vidB);
+
+    searchParams.delete(legacyParams.vidA);
+    searchParams.delete(legacyParams.vidB);
+
+    if (legacyA && uidA === '') {
+      searchParams.append(uidParams.vidA, UID_YT_NAMESPACE + legacyA);
+    }
+
+    if (legacyB && uidB === '') {
+      searchParams.append(uidParams.vidB, UID_YT_NAMESPACE + legacyB);
+    }
+
+    history.push('?' + searchParams.toString());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
Using `useEffect` still leads to unexpected side-effects. This effect runs after the first render, and `<VideoSelector>` components may trigger URLs rewrite themselves. 

To avoid this complexity, I would recommend to handle this redirection as soon as possible, for example by returning a react-router `<Redirect>` on the first render, if legacy params need to be rewritten.

Something like:
```
if (hasLegacyParams) {
  const newParams = rewriteParams()
  return (
    <Redirect
      to={{ pathname: location.pathname, search: newParams.toString() }}
    />
  );
}
```

Reply:
Sure, I switched from `useEffect` to a redirection and removed unnecessary code in the last 4 commits.